src/algorithms/neal2_algorithm.cpp

```diff
@@ -29,7 +29,10 @@ void Neal2Algorithm::print_startup_message() const {
 void Neal2Algorithm::sample_allocations() {
   // Initialize relevant values
   unsigned int n_data = data.rows();
-  auto &rng = bayesmix::Rng::Instance().get();
+  int ndata_from_hier = 0;
+  for (auto &clus : unique_values) ndata_from_hier += clus->get_card();
+  assert(n_data == ndata_from_hier);
```

Comment:
I'd add a DEBUG flag to these lines or something, if not just flat-out delete them

Reply:
All the assert are disabled when passing the NDEBUG compiler flag. I would keep them as they are helpful for debugging

Reply:
What you're saying is fine for one-liners with assert, but this is not (and maybe some of the other ones I reported), since there is a for loop. I'd add the flag in those cases

Reply:
Nice i didn't know!